test/unit/annotation_spec.js

```diff
@@ -233,6 +233,74 @@ describe('annotation', function() {
     });
   });
 
+  describe('TextAnnotation', function() {
+    it('should not parse state model when not defined',
+        function (done) {
+      const annotationDict = new Dict();
+      annotationDict.set('Type', Name.get('Annot'));
+      annotationDict.set('Subtype', Name.get('Text'));
+      annotationDict.set('Contents', 'TestText');
+      const annotationRef = new Ref(819, 0);
+
+      const replyDict = new Dict();
+      replyDict.set('Type', Name.get('Annot'));
+      replyDict.set('Subtype', Name.get('Text'));
+      replyDict.set('IRT', annotationDict);
+      replyDict.set('RT', Name.get('R'));
+      replyDict.set('Contents', 'ReplyText');
+
+      const replyRef = new Ref(820, 0);
+
+      const xref = new XRefMock([
+        { ref: annotationRef, data: annotationDict, },
+        { ref: replyRef, data: replyDict, }
+      ]);
+
+      AnnotationFactory.create(xref, replyRef, pdfManagerMock,
+        idFactoryMock).then(({ data, }) => {
+          expect(data.state).toBeNull();
+          expect(data.stateModel).toBeNull();
+          done();
+        }, done.fail);
+    });
+
+    it('should correctly parse state model and state',
+        function (done) {
+      const annotationDict = new Dict();
+      annotationDict.set('Type', Name.get('Annot'));
+      annotationDict.set('Subtype', Name.get('Text'));
+      annotationDict.set('Contents', 'AnnotText');
+
+      const annotationRef = new Ref(819, 0);
+
+      const replyDict = new Dict();
+      replyDict.set('Type', Name.get('Annot'));
+      replyDict.set('Subtype', Name.get('Text'));
+      replyDict.set('IRT', annotationDict);
+      replyDict.set('RT', Name.get('R'));
+      replyDict.set('StateModel', 'Review');
+      replyDict.set('State', 'Rejected');
+      replyDict.set('Contents', 'ReplyText');
+
+      const replyRef = new Ref(820, 0);
+
+      const xref = new XRefMock([
+        { ref: annotationRef, data: annotationDict, },
+        { ref: replyRef, data: replyDict, }
+      ]);
+
+      AnnotationFactory.create(xref, replyRef, pdfManagerMock,
+        idFactoryMock).then(({ data, }) => {
+          expect(data.inReplyTo).toBeDefined();
```

Comment:
I would suggest to leave out this line and the one below it because this test only tests `stateModel` and `state` being set correctly. The other fields should be covered in other tests, which you already have below, to keep the test's scope limited.